Explorer/Assets/DCL/Multiplayer/Connections/Archipelago/SignFlow/LiveConnectionArchipelagoSignFlow.cs

```diff
@@ -0,0 +1,123 @@
+                    using var challenge = multiPool.TempResource<ChallengeRequestMessage>();
+                    challenge.value.Address = ethereumAddress;
+                    using var clientPacket = multiPool.TempResource<ClientPacket>();
+                    clientPacket.value.ClearMessage();
+                    clientPacket.value.ChallengeRequest = challenge.value;
+                    using var response = await connection.SendAndReceiveAsync(clientPacket.value, memoryPool, token);
+                    using var serverPacket = new SmartWrap<ServerPacket>(response.AsMessageServerPacket(), multiPool);
+                    using var challengeResponse = new SmartWrap<ChallengeResponseMessage>(serverPacket.value.ChallengeResponse!, multiPool);
+                    return challengeResponse.value.ChallengeToSign!;
+                },
+                cancellationToken: token
+            );
+
+        public UniTask<LightResult<string>> WelcomePeerIdAsync(string signedMessageAuthChainJson, CancellationToken token) =>
+            UniTask.RunOnThreadPool(async () =>
+                {
+                    using var signedMessage = multiPool.TempResource<SignedChallengeMessage>();
+                    signedMessage.value.AuthChainJson = signedMessageAuthChainJson;
+
+                    using var clientPacket = multiPool.TempResource<ClientPacket>();
+                    clientPacket.value.ClearMessage();
+                    clientPacket.value.SignedChallenge = signedMessage.value;
+
+                    var linkedToken = CancellationTokenSource.CreateLinkedTokenSource(new CancellationTokenSource().Token, token);
+
+                    var result = await UniTask.WhenAny(
+                        connection.SendAndReceiveAsync(clientPacket.value, memoryPool, linkedToken.Token),
+                        connection.WaitDisconnectAsync(linkedToken.Token)
+                    );
+
+                    linkedToken.Cancel();
+
+                    if (result.hasResultLeft)
+                    {
+                        using var response = result.result;
+                        using var serverPacket = new SmartWrap<ServerPacket>(response.AsMessageServerPacket(), multiPool);
+                        using var welcomeMessage = new SmartWrap<WelcomeMessage>(serverPacket.value.Welcome!, multiPool);
+                        return welcomeMessage.value.PeerId.AsSuccess();
+                    }
+
+                    return LightResult<string>.FAILURE;
+                },
+                cancellationToken: token
+            );
+
+        public UniTask SendHeartbeatAsync(Vector3 playerPosition, CancellationToken token) =>
+            UniTask.RunOnThreadPool(
+                () =>
+                {
+                    using var position = multiPool.TempResource<Position>();
+                    position.value.X = playerPosition.x;
+                    position.value.Y = playerPosition.y;
+                    position.value.Z = playerPosition.z;
```

Comment:
bite-compression of it is out of the scope of this PR, right?

Reply:
It's not the case, Archipelago already has established API, we won't use it via livekit